astar_utils/nested_mapping.py

```diff
@@ -28,6 +28,9 @@ def __init__(self, new_dict: Optional[Iterable] = None,
                 self.update(entry)
 
     def update(self, new_dict: MutableMapping[str, Any]) -> None:
+        if isinstance(new_dict, self.__class__):
```

Comment:
True. Didn't consider that. "don't spell the class name in the class" gone too far...

Reply:
Now I wonder whether it is even possible to refer to the `NestedMapping` here without typing it out. Probably through introspection, because the function knows in which class it is defined.

Reply:
Hm, I thought it might be possible via `new_dict.__class__ in self.__class__.__mro__` for subclasses. But after running a simplified version of something like that, it seems the `__class__` is not considered equal even if it is when converted to `str`. Weird. But anyway, for this case it's fine to just spell out `NestedMapping` I guess 🙃